odoo/tools/cache.py

```diff
@@ -63,8 +63,12 @@ def __call__(self, method):
     def determine_key(self):
         """ Determine the function that computes a cache key from arguments. """
         if self.skiparg is None:
+            # Remove annotations because lambdas can't be type-annotated
+            args = ', '.join(
+                str(params.replace(annotation=Parameter.empty))
```

Comment:
What about getting rid of the default as well?
As far as I can tell, it is not problematic, but is actually useless in this case, because the "defaulting" process already happens in the wrapper function.
```suggestion
                str(params.replace(annotation=Parameter.empty, default=Parameter.empty))
```
Maybe I am overthinking this stuff, but could the default be problematic if the value is not `repr`esentable?

Reply:
It doesn't seem correct. 
if you take one of our example: 
```python
    @api.model
    @tools.ormcache('path', 'query_args')
    def url_rewrite(self, path, query_args=None):
```

The lambda result, will be `lambda path, query_args: (path, query_args)` instead of `lambda path, query_args=None: (path, query_args)`.
And if you call the method with only one arg, it will crash (I suppose, I didn't test) instead of resulting a tuple `(path, None)` as key.

Reply:
I just checked what `decorator` does: it reproduces the API of the wrapped method, which includes the defaulting process.  As the lambda is called from inside the wrapper, all defaults are there already.  That's why they are redundant inside the lambda.

Reply:
Yes, you are perfectly right. I updated the code and add a comment about it.

>  could the default be problematic if the value is not representable?

I tested it, default value is evaluated before the lambda creation then it is not possible to have this issue, but it is still redundant.